File: 5. Right Triangle Patterns (Bottom-Left)/44. Right Triangle BL Decreasing Numbers/bldn.py
n = int(input("Enter a Number : "))

for i in range(1,n+1):
    for j in range(n,i-1,-1): # for j in range(1,n-i+2):
        print(j,end=" ")
    print()

'''

Use (i+j<=n+1) for optimized code & for n matrix 

Output:

5 4 3 2 1 
5 4 3 2 
5 4 3 
5 4 
5  
 
'''

File: Patterns/05. Right Triangle Patterns (Bottom-Left)/44. Right Triangle BL Decreasing Numbers/bldn.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=n; i++){
        for(int j=i; j<=n; j++){ // for(int j=1; j<=(n-i+1); j++)
            cout<<n+i-j<<" ";
        }
        cout<<'\n';
    }
}

/*

Use (i+j<=n+1) for optimized code & for n matrix 

Output

5 4 3 2 1 
5 4 3 2 
5 4 3 
5 4 
5 

*/